homeassistant/components/airvisual/__init__.py

```diff
@@ -167,35 +160,70 @@ async def async_setup_entry(hass, config_entry):
 
     if CONF_API_KEY in config_entry.data:
         _standardize_geography_config_entry(hass, config_entry)
-        airvisual = AirVisualGeographyData(
+
+        client = Client(api_key=config_entry.data[CONF_API_KEY], session=websession)
+
+        async def async_update_data():
+            """Get new data from the API."""
+            if CONF_CITY in config_entry.data:
+                return await client.api.city(
+                    config_entry.data[CONF_CITY],
+                    config_entry.data[CONF_STATE],
+                    config_entry.data[CONF_COUNTRY],
+                )
```

Comment:
It looks like this changed to return directly instead of setting it as `api_coro` and passing it though the try.  Was that intentional? Can this fail in the same way as `nearest_city` below?

Reply:
Great catch!